Examine Tooltip for Players Persists Thru Walls

Superlagg opened the pull request:
# Description

When you examine someone, the tooltip should still show up even if they hide behind a wall. Still obeys max distance though, so it'll disappear if they run too far away.

Whole point of this is to make it a bit easier to read peoples' flavortexts.

---

# Changelog

:cl:
- tweak: Player examine tooltips will persist even if they go out of line of sight.

Superlagg committed:
Examine tooltip for players persists thru walls

Mnemotechnician commented on Content.Shared/Examine/ExamineSystemShared.cs:
```diff
@@ -109,6 +110,11 @@ public virtual bool CanExamine(EntityUid examiner, MapCoordinates target, Ignore
             if (EntityManager.GetComponent<TransformComponent>(examiner).MapID != target.MapId)
                 return false;
 
+            // if the examined thing is a player, just return true
```
```suggestion
            // Floofstation edit - if the examined thing is a player, just return true
```

Mnemotechnician commented on Content.Shared/Examine/ExamineSystemShared.cs:
```diff
@@ -320,9 +326,6 @@ public ExaminedEvent(FormattedMessage message, EntityUid examined, EntityUid exa
 
-        /// <summary>
-        ///     Returns <see cref="Message"/> with all <see cref="Parts"/> appended according to their priority.
-        /// </summary>
```
Why is this getting removed?

Superlagg replied:
Was a mistake, putting it back!

Mnemotechnician commented:
Also, I do not see a range check here. You are skipping the entire InRangeUnoccluded call, but not providing a fallback range check.

Superlagg committed:
Update Content.Shared/Examine/ExamineSystemShared.cs

Co-authored-by: Mnemotechnican <69920617+Mnemotechnician@users.noreply.github.com>

Superlagg commented:
> Also, I do not see a range check here. You are skipping the entire InRangeUnoccluded call, but not providing a fallback range check.

That's what I thought, but in testing, the tooltip still disappeared after moving a screen away from the dummy I gave an actor component. Not sure why that is, though.

Mnemotechnician commented:
I would prefer not to rely on undefined behavior and add an explicit range check here. You can achieve it by adding a simple `&& Tranform(examiner).TryDistance(EntityManager, _tranform, Tranform(target).Coordinates, out var distance) <= GetExaminerRange(examiner)` clause to your `if` statement's condition, or otherwise.

Superlagg committed:
Restores comment I removed for some reason

Superlagg committed:
Merge branch 'your-description-of-your-foxboys-ass-will-not-evade-MY-eyes' of https://github.com/Superlagg/floofstation1 into your-description-of-your-foxboys-ass-will-not-evade-MY-eyes

Superlagg committed:
Adds a distance check to the wall-ignoring examine tooltip
Still seems to require the mob to be within your screen area, albeit they can be behind walls.

Mnemotechnician commented on Content.Shared/Examine/ExamineSystemShared.cs:
```diff
@@ -109,6 +110,14 @@ public virtual bool CanExamine(EntityUid examiner, MapCoordinates target, Ignore
 
+            // Floofstation edit - if the examined thing is a player, just return true
+            if (examined != null)
+                if (EntityManager.TryGetComponent<ActorComponent>(examined, out var _)
```
```suggestion
                if (TryComp<ActorComponent>(examined, out var _)
```

Superlagg replied:
Sure thing! I assume TryComp just checks if the component exists, while TryGetComponent also fetches the actual component?

Mnemotechnician replied:
It is the same thing, but in the more preferable way

Superlagg committed:
Update Content.Shared/Examine/ExamineSystemShared.cs

Co-authored-by: Mnemotechnican <69920617+Mnemotechnician@users.noreply.github.com>
